### PR TITLE
Add oracles to credit calculations

### PR DESCRIPTION
**Pull request by vanruch**
This PR converts `tvl` in TRA and `borrowLimit` and `totalBorrowed` in TCA to USD with 18 decimal points. 

This will result in a few important changes:

- Borrow limit will return USD amount, but later borrowers will try to borrow amounts in other assets which can be confusing espetially with large price fluctuations
- Since borrowLimit is in USD now, the loans in non-stable-USD asstets might go above borrow limit when price rises which raises questions - what should happen in this case?

Maybe we want credit lines in different assets to be tracked completely separately (i.e. different maxBorrowLimits in oracle, different pools accounting to tvl)?

### CODE REVIEW

**Review by yuchenlintt — COMMENTED** [review]
LGTM after addressing @skalermo's comments

File: contracts/truefi2/TrueCreditAgency.sol [hunk 1]
```diff
@@ -266,22 +266,13 @@ contract TrueCreditAgency is UpgradeableClaimable, ITrueCreditAgency {
...
-     * @param decimals Precision to use when calculating total borrowed
-     * @return Total amount borrowed for `borrower` in USD
+     * @return borrowSum Total amount borrowed for `borrower` in USD
      */
-    function totalBorrowed(address borrower, uint8 decimals) public view returns (uint256) {
-        uint256 borrowSum = 0;
-        uint256 resultPrecision = uint256(10)**decimals;
-
-        // loop through pools and sum amount borrowed accounting for precision
+    function totalBorrowed(address borrower) public view returns (uint256 borrowSum) {
```
[the remark this answers is not in the document]
Comment (skalermo) [reply contracts/truefi2/TrueCreditAgency.sol:271]: Oh, really? I must have forgotten. If so, then we can consider this one as resolved.

File: contracts/truefi2/TrueCreditAgency.sol [hunk 2]
```diff
@@ -266,22 +266,13 @@ contract TrueCreditAgency is UpgradeableClaimable, ITrueCreditAgency {
...
+        // loop through pools and sum amount borrowed converted to USD
         for (uint8 i = 0; i < pools.length; i++) {
```
[the remark this answers is not in the document]
Comment (vanruch) [reply contracts/truefi2/TrueCreditAgency.sol:274]: This will never fit into a gas limit

[the remark this answers is not in the document]
Comment (skalermo) [reply contracts/truefi2/TrueCreditAgency.sol:274]: Ok, I get it now. And even 255 is too many because of the gas consumption. Marking this as resolved.

**Review by skalermo — APPROVED** [review]

**Review by yuchenlintt — APPROVED** [review]

**Pull request merged**